File: 백준/Silver/10866. 덱/덱.py
import sys
from collections import deque

N = int(sys.stdin.readline())

queue = deque()

for i in range(N):
    input_arr = sys.stdin.readline().split()
    command = input_arr[0]
    if command == "push_back":
        value = int(input_arr[1])
        queue.append(int(value))
    elif command == "push_front":
        value = int(input_arr[1])
        queue.appendleft(int(value))
    else:
        if command == "pop_front":
            if len(queue) == 0:
                print(-1)
            else:
                pop_value = queue.popleft()
                print(pop_value)
        elif command == "pop_back":
            if len(queue) == 0:
                print(-1)
            else:
                pop_value = queue.pop()
                print(pop_value)
        elif command == "size":
            print(len(queue))
        elif command == "empty":
            if len(queue) == 0:
                print(1)
            else:
                print(0)
        elif command == "front":
            if len(queue) == 0:
                print(-1)
            else:
                print(f"{queue[0]}")
        elif command == "back":
            size = len(queue)
            if size == 0:
                print(-1)
            else:
                print(f"{queue[size - 1]}")

File: 백준/Silver/10866. 덱/덱.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());
        Deque<Integer> q = new ArrayDeque<>();
        StringBuilder answer = new StringBuilder();

        for (int i = 0; i < n; i++) {
            String[] input = br.readLine().split(" ");
            switch (input[0]) {
                case "push_back":
                    q.addLast(Integer.parseInt(input[1]));
                    break;
                case "push_front":
                    q.addFirst(Integer.parseInt(input[1]));
                    break;
                case "pop_front":
                    if (q.isEmpty()) {
                        answer.append("-1\n");
                    } else {
                        answer.append(q.pollFirst() + "\n");
                    }
                    break;
                case "pop_back":
                    if (q.isEmpty()) {
                        answer.append("-1\n");
                    } else {
                        answer.append(q.pollLast() + "\n");
                    }
                    break;
                case "size":
                    answer.append(q.size() + "\n");
                    break;
                case "empty":
                    if (q.isEmpty()) {
                        answer.append("1\n");
                    } else {
                        answer.append("0\n");
                    }
                    break;
                case "front":
                    if (q.isEmpty()) {
                        answer.append("-1\n");
                    } else {
                        answer.append(q.peekFirst() + "\n");
                    }
                    break;
                case "back":
                    if (q.isEmpty()) {
                        answer.append("-1\n");
                    } else {
                        answer.append(q.peekLast() + "\n");
                    }
                    break;
            }
        }
        bw.write(answer.toString());
        br.close();
        bw.close();
    }
}
